<a href="https://colab.research.google.com/github/rynedaniels/food-sales-predictions/blob/main/Project1_Part_2_Charles_DanielsB.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Imports

In [23]:
import pandas as pd
import numpy as np


# Loading Data #

In [24]:
df = pd.read_csv('/content/drive/MyDrive/Coding Dojo/02 Week 2: Pandas/sales_predictions.csv')
df.head()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


# **Data Cleaning**| #


## Determining data shape:
#### How many rows and columns?

In [25]:
df.shape

(8523, 12)

- There are 8523 rows and 12 columns

## Checking the data types:
#### What are the datatypes of each variable?

In [26]:
df.dtypes

Item_Identifier               object
Item_Weight                  float64
Item_Fat_Content              object
Item_Visibility              float64
Item_Type                     object
Item_MRP                     float64
Outlet_Identifier             object
Outlet_Establishment_Year      int64
Outlet_Size                   object
Outlet_Location_Type          object
Outlet_Type                   object
Item_Outlet_Sales            float64
dtype: object

## Checking for duplicates:
#### Are there duplicates? If so, drop any duplicates

In [27]:
df.duplicated().sum()

0

- There are zero duplicates in our datset

## Identifying any missing values:
#### Are there any missing values in our dataset?

In [28]:
df.isna().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

- The `Item_Weight` column is missing `1463` values. **(17.2% of values)**
- The `Outlet_Size` column is missing `2410` values. **(28.8 % of values)**

## Addressing the Missing Values
#### Deciding on how to address the missing values

- Considering the name of the dataset is "sales_predictions." I conclude that neither the `item_weight` or `outlet_size` columns will not be a key determining factors, so both will be dropped.




In [34]:
df2 = df.drop(columns= ['Item_Weight', 'Outlet_Size'])
df2

Item_Identifier Item_Fat_Content  Item_Visibility              Item_Type  \
0              FDA15          Low Fat         0.016047                  Dairy   
1              DRC01          Regular         0.019278            Soft Drinks   
2              FDN15          Low Fat         0.016760                   Meat   
3              FDX07          Regular         0.000000  Fruits and Vegetables   
4              NCD19          Low Fat         0.000000              Household   
...              ...              ...              ...                    ...   
8518           FDF22          Low Fat         0.056783            Snack Foods   
8519           FDS36          Regular         0.046982           Baking Goods   
8520           NCJ29          Low Fat         0.035186     Health and Hygiene   
8521           FDN46          Regular         0.145221            Snack Foods   
8522           DRG01          Low Fat         0.044878            Soft Drinks   

      Item_MRP Outlet_Identifier  Outlet_Establishment_Year  \
0     249.8092            OUT049                       1999   
1      48.2692            OUT018                       2009   
2     141.6180            OUT049                       1999   
3     182.0950            OUT010                       1998   
4      53.8614            OUT013                       1987   
...        ...               ...                        ...   
8518  214.5218            OUT013                       1987   
8519  108.1570            OUT045                       2002   
8520   85.1224            OUT035                       2004   
8521  103.1332            OUT018                       2009   
8522   75.4670            OUT046                       1997   

     Outlet_Location_Type        Outlet_Type  Item_Outlet_Sales  
0                  Tier 1  Supermarket Type1          3735.1380  
1                  Tier 3  Supermarket Type2           443.4228  
2                  Tier 1  Supermarket Type1          2097.2700  
3                  Tier 3      Grocery Store           732.3800  
4                  Tier 3  Supermarket Type1           994.7052  
...                   ...                ...                ...  
8518               Tier 3  Supermarket Type1          2778.3834  
8519               Tier 2  Supermarket Type1           549.2850  
8520               Tier 2  Supermarket Type1          1193.1136  
8521               Tier 3  Supermarket Type2          1845.5976  
8522               Tier 1  Supermarket Type1           765.6700  

[8523 rows x 10 columns]